### ISSUE TITLE
Make Frontend CLI work with Delegate server

### ISSUE THREAD

**Issue by uhbif19** [mlabs-haskell/hydra-auction#205]
How CLI should work is covered by relevant section of off_chain_spec, while problems might occur.

1.

The first obstacle I see is tracking a delegate server/head state: probably it would be streamed by delegate server and should be recorded by listener thread. There is at least standing bid state and HeadId.

HeadId we should get from Delegate server before announcing auction. So CLI may probably just wait for it before starting or fail to do announce-auction command.

Current idea is to implement that is to update from `DelegateState` pushes from Delegate server. So that should probably be separate thread which waits for `DelegateState` and updates accordingly. That should be working the same in both Watch and Repl interfaces.

2.

Second issue is tracking L1/L2 status (and its possible interference with Auction stages). This should be tracked

I thing that we should introduce whole auction state state-machine datatype and store it in variable. It should keep track of both if standing bid is in L1 or L2 **and** the `DelegateState` from point 1 (obviously this is only applicable for L1 case).

So it would probably look like:

```
data AuctionState =
     NotAnnounced ...
     | OnL2 DelegateState
     | OnL1 ...
```

Tasks (draft):

- [x] L2 version of StartBidding action (see spec)
- [x] Tracking of L1/L2 status of bidding
- [x] Placing bid on L2 (see spec)
- [ ] Validate delegate state in CLI
- [ ] Cover possible problems with L1/L2 moving
- [ ] Changing CLI E2E test to check all of that.

Closing Head on L1 is out of scope of this task and in #160 

**UPD:** Separated E2E CLI tests to #239

Blocked by #175 
Blocked by #239
Blocked by #204
